scripts/ci/release-github-actions.sh

```diff
@@ -1,139 +1,267 @@
+            if [ "${diff}" -gt 0 ]; then
+                echo "Moving tag ${TAG_NAME} to new release ${RELEASE_TAG}"
+                if [ "$DRY_RUN" = false ]; then
+                    git tag -f "${TAG_NAME}" "${RELEASE_TAG}"
+                else
+                    echo "Dry run: Skipping moving tag ${TAG_NAME}"
+                fi
+            else
+                echo "Existing ${TAG_NAME} tag points to an equal or newer version. Skipping."
+            fi
+        else
+            echo "Creating tag ${TAG_NAME} for release ${RELEASE_TAG}"
+            if [ "$DRY_RUN" = false ]; then
+                git tag "${TAG_NAME}" "${RELEASE_TAG}"
+            else
+                echo "Dry run: Skipping creating tag ${TAG_NAME}"
+            fi
         fi
+    }
 
-        VERSIONS_BUCKET_FILENAME="downloads.okteto.com/cli/stable/versions"
+    # Update MAJOR_TAG
+    update_tag_if_greater "${MAJOR_TAG}"
 
-        VERSIONS_FILE=$(mktemp)
-        gsutil cat "gs://${VERSIONS_BUCKET_FILENAME}" >"$VERSIONS_FILE"
-        echo "Current version list for stable channel (showing latest 10):"
-        tail -n 10 "$VERSIONS_FILE"
+    # Update MINOR_TAG
+    update_tag_if_greater "${MINOR_TAG}"
 
-        if ! grep -qFx "${RELEASE_TAG}" "${VERSIONS_FILE}"; then
-                echo "RELEASE_TAG: ${RELEASE_TAG} not included in the versions from https://${VERSIONS_BUCKET_FILENAME}"
-                exit 1
+    # Create release branch
+    if git rev-parse --verify --quiet "${RELEASE_BRANCH}"; then
+        echo "Release branch ${RELEASE_BRANCH} already exists. Skipping branch creation."
+    else
+        echo "Creating release branch ${RELEASE_BRANCH}"
+        if [ "$DRY_RUN" = false ]; then
+            git branch "${RELEASE_BRANCH}" "${RELEASE_TAG}"
+        else
+            echo "Dry run: Skipping creation of release branch ${RELEASE_BRANCH}"
         fi
+    fi
 
-        repos=(
-                delete-namespace
-                build
-                destroy-preview
-                deploy-preview
-                deploy-stack
-                namespace
-                pipeline
-                push
-                create-namespace
-                destroy-pipeline
-                login
-                destroy-stack
-                apply
-                context
-                test
-        )
-
-        for repo in "${repos[@]}"; do
-                echo "Releasing ${REPO_OWNER}/$repo"
-                git clone "git@github.com:${REPO_OWNER}/${repo}.git"
-                pushd "$repo"
-                git config user.name "okteto"
-                git config user.email "ci@okteto.com"
-
-                if git tag --list | grep -qFx "$RELEASE_TAG"; then
-                        echo "$RELEASE_TAG already exists"
-                        exit 1
-                fi
+    # Update 'latest' and 'stable' tags if RELEASE_TAG is greater
+    update_tag_if_greater "latest"
+    update_tag_if_greater "stable"
```

Comment:
This has to be always updated, right? Also, how does `latest` and `stable` compare to a semantic version (which is the operation done within `update_tag_if_greater`)?

Reply:
We are not comparing the `latest` and `stable` with a semver.

I will try to explain the function `update_tag_if_greater` the best I can. Also I will add some comments in order to make it more explainable:

1. Check if the tag already exists

> if git rev-parse "${TAG_NAME}" >/dev/null 2>&1; then

We check if latest or stable exists as a tag

2. Getting the last official stable version

> current_tag_version=$(git describe --tags "${TAG_NAME}" --abbrev=0 --match "*.*.*" 2>/dev/null || echo "0.0.0")

This command is to retrieve the semver associated with a specific Git tag (TAG_NAME). If it fails to find such a version, it defaults to `0.0.0`. For example in the following screenshot, if you pass latest as argument we return `1.21.3`
<img width="635" alt="Screenshot 2024-10-03 at 13 41 11" src="https://github.com/user-attachments/assets/fb237a95-74a2-478c-a56d-a2cd216e956a">

3. Compare with the version that we got in previous command:

> diff=$(semver compare "${RELEASE_TAG}" "${current_tag_version}")

We check if the release we are releasing is greater than the one is set as latest.

Reply:
Ah, Ok, I had missed the command to retrieve the semantic version from the `latest` or `stable` tag, thanks for the explanation